File: src/my_cpp_pkg/src/robot_news_station.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/string.hpp"

class RobotNewsStationNode: public rclcpp::Node
{
    public:
        RobotNewsStationNode():Node("robot_new_station"),robot_name_("hash")
        {
            RCLCPP_INFO(this->get_logger(),"Robot news station node created");
            timer_ = this->create_wall_timer(std::chrono::milliseconds(500),std::bind(&RobotNewsStationNode::timerCallback,this));
            publisher_ = this->create_publisher<example_interfaces::msg::String>("robot_news",10);
        }
    private:
        void timerCallback()
        {
            auto msg = example_interfaces::msg::String();
            msg.data = std::string("Hi this is ")+robot_name_+std::string(" from Mumbai");
            publisher_->publish(msg);
        }
        rclcpp::TimerBase::SharedPtr timer_;
        rclcpp::Publisher<example_interfaces::msg::String>::SharedPtr publisher_;
        std::string robot_name_;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc,argv);
    auto node = std::make_shared<RobotNewsStationNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: src/my_py_pkg/my_py_pkg/robot_news_station.py
import rclpy
from rclpy.node import Node

from example_interfaces.msg import String

class RobotNewsStationNode(Node):

    def __init__(self):
        super().__init__("robot_news_station")
        self.get_logger().info("Created robot_news_station Node")

        self.publisher_ = self.create_publisher(String, "robot_news", 10)

        self.timer_ = self.create_timer(0.5,self.publish_news)

    def publish_news(self):
        msg = String()
        msg.data = "Hello"
        self.publisher_.publish(msg)

def main(args=None):
    rclpy.init(args=args)
    node = RobotNewsStationNode()
    rclpy.spin(node)
    rclpy.shutdown()


if __name__ == "__main__":
    main()